File: src/KDDockWidgets.cpp
/*
  This file is part of KDDockWidgets.

  SPDX-FileCopyrightText: 2019 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
  Author: Sérgio Martins <sergio.martins@kdab.com>

  SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only

  Contact KDAB at <info@kdab.com> for commercial licensing options.
*/

#include "Config.h"
#include "kddockwidgets/KDDockWidgets.h"
#include "core/layouting/Item_p.h"

#ifdef KDDW_FRONTEND_QTWIDGETS
#include "qtwidgets/Platform.h"
#endif

#ifdef KDDW_FRONTEND_QTQUICK
#include "qtquick/Platform.h"
#endif

using namespace KDDockWidgets;

#ifdef KDDW_FRONTEND_QT

void KDDockWidgets::initFrontend(FrontendType type)
{
    if (Core::Platform::instance())
        return;

    switch (type) {
    case FrontendType::QtWidgets:
#ifdef KDDW_FRONTEND_QTWIDGETS
        new QtWidgets::Platform();
#endif
        break;
    case FrontendType::QtQuick:
#ifdef KDDW_FRONTEND_QTQUICK
        new QtQuick::Platform();
#endif
        break;
    case FrontendType::Flutter:
        // Nothing to do, called from Dart
        break;
    }
}

#endif

const char *KDDockWidgets::spdlogLoggerName()
{
    return "com.kdab.dockwidgets";
}

NeighbourSqueezeStrategy InitialOption::s_defaultNeighbourSqueezeStrategy = NeighbourSqueezeStrategy::AllNeighbours;

InitialOption::InitialOption()
{
}

InitialOption::InitialOption(InitialVisibilityOption v)
    : visibility(v)
{
}

InitialOption::InitialOption(QSize size)
    : preferredSize(size)
{
}

InitialOption::InitialOption(InitialVisibilityOption v, QSize size)
    : visibility(v)
    , preferredSize(size)
{
}

InitialOption::InitialOption(DefaultSizeMode mode)
    : sizeMode(mode)
{
}

File: src/flutter/dart/lib/KDDockWidgets.dart
/*
  This file is part of KDDockWidgets.

  SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
  Author: Sérgio Martins <sergio.martins@kdab.com>

  SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only

  Contact KDAB at <info@kdab.com> for commercial licensing options.
*/

library kddockwidgets;

import 'package:KDDockWidgets/models/GeometryItem.dart';
import 'package:KDDockWidgets/models/Separator.dart';
import 'package:KDDockWidgets/private/Bindings.dart';
import 'package:flutter/material.dart';

import 'dart:ffi' as ffi;

import 'package:signals_slots/signals_slots.dart';

part 'models/Group.dart';
part 'models/DropArea.dart';
part 'models/TitleBar.dart';
part 'models/FloatingItem.dart';
part 'models/DockRegistry.dart';
part 'models/DockItem.dart';

class DropLocation {
  static const int DropLocation_None = 0;
  static const int DropLocation_Left = 1;
  static const int DropLocation_Top = 2;
  static const int DropLocation_Right = 4;
  static const int DropLocation_Bottom = 8;
  static const int DropLocation_Center = 16;
  static const int DropLocation_OutterLeft = 32;
  static const int DropLocation_OutterTop = 64;
  static const int DropLocation_OutterRight = 128;
  static const int DropLocation_OutterBottom = 256;
  static const int DropLocation_Inner = DropLocation_Left |
      DropLocation_Right |
      DropLocation_Top |
      DropLocation_Bottom;
  static const int DropLocation_Outter = DropLocation_OutterLeft |
      DropLocation_OutterRight |
      DropLocation_OutterTop |
      DropLocation_OutterBottom;
  static const int DropLocation_Horizontal = DropLocation_Left |
      DropLocation_Right |
      DropLocation_OutterLeft |
      DropLocation_OutterRight;
  static const int DropLocation_Vertical = DropLocation_Top |
      DropLocation_Bottom |
      DropLocation_OutterTop |
      DropLocation_OutterBottom;
}
